src/server/pkg/storage/chunk/storage.go

```diff
@@ -41,30 +44,48 @@ func NewStorage(objC obj.Client, opts ...StorageOption) *Storage {
 
 // NewReader creates a new Reader.
 func (s *Storage) NewReader(ctx context.Context, dataRefs ...*DataRef) *Reader {
-	return newReader(ctx, s.objC, dataRefs...)
+	return newReader(ctx, s.objClient, dataRefs...)
 }
 
 // NewWriter creates a new Writer for a stream of bytes to be chunked.
 // Chunks are created based on the content, then hashed and deduplicated/uploaded to
 // object storage.
 // The callback arguments are the chunk hash and annotations.
-func (s *Storage) NewWriter(ctx context.Context, averageBits int, seed int64, noUpload bool, f WriterFunc) *Writer {
-	return newWriter(ctx, s.objC, averageBits, f, seed, noUpload)
+func (s *Storage) NewWriter(ctx context.Context, averageBits int, seed int64, noUpload bool, tmpID string, f WriterFunc) *Writer {
+	return newWriter(ctx, s.objClient, s.gcClient, averageBits, f, seed, noUpload, tmpID)
 }
 
 // List lists all of the chunks in object storage.
 func (s *Storage) List(ctx context.Context, f func(string) error) error {
-	return s.objC.Walk(ctx, prefix, f)
+	return s.objClient.Walk(ctx, prefix, f)
 }
 
 // DeleteAll deletes all of the chunks in object storage.
 func (s *Storage) DeleteAll(ctx context.Context) error {
-	return s.objC.Walk(ctx, prefix, func(hash string) error {
-		return s.objC.Delete(ctx, hash)
+	return s.objClient.Walk(ctx, prefix, func(hash string) error {
+		return s.objClient.Delete(ctx, hash)
 	})
 }
 
 // Delete deletes a chunk in object storage.
 func (s *Storage) Delete(ctx context.Context, hash string) error {
-	return s.objC.Delete(ctx, path.Join(prefix, hash))
+	return s.objClient.Delete(ctx, path.Join(prefix, hash))
+}
+
+// AddSemanticReference adds a semantic reference to a chunk.
+func (s *Storage) AddSemanticReference(ctx context.Context, name string, chunk *Chunk) error {
+	return s.gcClient.AddReference(ctx, semanticReference(name, chunk.Hash))
+}
+
+// RemoveSemanticReference removes a semantic reference.
+func (s *Storage) RemoveSemanticReference(ctx context.Context, name string) error {
+	return s.gcClient.RemoveReference(ctx, semanticReference(name, ""))
+}
+
+func semanticReference(name, chunk string) *gc.Reference {
+	return &gc.Reference{
+		Sourcetype: "semantic",
+		Source:     name,
+		Chunk:      path.Join(prefix, chunk),
+	}
```

Comment:
Could we use `Create` and `Delete` here in place of `Add` and `Remove`? Those seem more consistent with the rest of our APIs and afaict don't have different meanings.

Reply:
Sure.